src/d3a/models/state.py

```diff
@@ -43,131 +46,249 @@
 # - If a device has no state, maybe it doesn't need its own appliance class either
 
 
-class PVState:
+class StateInterface(metaclass=ABCMeta):
+    """Interface containing methods that need to be defined by each State class."""
+
+    @abstractmethod
+    def get_state(self) -> Dict:
+        """Return the current state of the device."""
+        return {}
```

Comment:
The implementation here should be removed as it's an abstract function

Reply:
I've replied here: https://github.com/gridsingularity/d3a/pull/1159#discussion_r653696953